src/zk.py

```diff
@@ -106,6 +101,19 @@ def __init__(self, config, plugins):
             for line in traceback.format_exc().split('\n'):
                 logging.error(line.rstrip())
 
+
+    def get_lock_contender_name(self):
+        if self._lock_contender_name:
+            return self._lock_contender_name
+        return helpers.get_hostname()
+
+
+    def _get_host_path(self, path, hostname):
+        if hostname is None:
+            hostname = self.get_lock_contender_name()
```

Comment:
I think it is not what we need.
`_get_host_path` was meant to manipulate with `path` template and put there a hostname.
Nothing to do with locks and contender names here.